### PR TITLE
Add other error item

### LINKED ISSUE #237: Add Other Item for Ics02Client,Ics03connection, Ics04Channel Error
## Summary

Add Other Item for Ics02Client,Ics03connection, Ics04Channel Error

## Problem Definition

For downstream implementations of ibc-rs in ics02-client, ics03-connection, ics04-channel, some third-party libraries need to have their own error output, and these modules do not have an error item compatible with third-party implementations. I suggest adding an Other item to the error handling of these modules for downstream implementations to use.

## Proposal

Add ics02-client, ics03-connection, ics04-channel Error module add below Other item.

```rust
define_error! {
    #[derive(Debug, PartialEq, Eq)]
    Error {
        Other 
            { description: String }
            | e | { format_args!("{}", e.description) }
   }
}
```
## Acceptance Criteria

____

#### For Admin Use

- [ ] Not duplicate issue
- [ ] Appropriate labels applied
- [ ] Appropriate milestone (priority) applied
- [ ] Appropriate contributors tagged
- [ ] Contributor assigned/self-assigned

### PR DESCRIPTION
**Pull request by DaviRain-Su**
Closes: #237 

## Description

______

### PR author checklist:
- [ ] Added changelog entry, using [`unclog`](https://github.com/informalsystems/unclog).
- [ ] Added tests.
- [ ] Linked to GitHub issue.
- [ ] Updated code comments and documentation (e.g., `docs/`).
- [ ] Tagged *one* reviewer who will be the one responsible for shepherding this PR.

### Reviewer checklist:

- [ ] Reviewed `Files changed` in the GitHub PR explorer.
- [ ] Manually tested (in case integration/unit/mock tests are absent).

### CODE REVIEW

File: crates/ibc/src/core/ics04_channel/error.rs [hunk 1]
```diff
@@ -351,6 +351,10 @@ define_error! {
         AbciConversionFailed
             { abci_event: String }
             | e | { format_args!("Failed to convert abci event to IbcEvent: {}", e.abci_event)},
+
+        Other
```
Comment (plafer) [inline crates/ibc/src/core/ics04_channel/error.rs:355]: We already have the [`AppModule` error](https://github.com/cosmos/ibc-rs/blob/0c5eb875f77f0de3360e58db957e8814aabbee60/crates/ibc/src/core/ics04_channel/error.rs#L343) for IBC applications.

What is the intended use of this error?

File: crates/ibc/src/core/ics03_connection/error.rs [hunk 2]
```diff
@@ -166,5 +166,9 @@ define_error! {
                 reason: String,
             }
             | e | { format_args!("invalid client state: {0}", e.reason) },
+
+        Other
```
Comment (plafer) [inline crates/ibc/src/core/ics03_connection/error.rs:170]: What is an example use case for this error? Users are not expected to generate any IBC core errors. Same comment applies to the client error.

File: crates/ibc/src/core/ics04_channel/error.rs (hunk 1 above)
Comment (plafer) [reply crates/ibc/src/core/ics04_channel/error.rs:355]: Note: We also have the [`ImplementationSpecific` error](https://github.com/cosmos/ibc-rs/blob/249650d0c396f16e6bdc43ae9e5fb9892580082f/crates/ibc/src/core/ics02_client/error.rs#L49) for clients, which I believe serves the same purpose as `Other` here... But I prefer `Other` as a name actually!

File: crates/ibc/src/core/ics03_connection/error.rs (hunk 2 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics03_connection/error.rs:170]: Here's my specific code use case.
Here I'm deserializing the data to get other types of errors in the ibc-rs interface implementation layer, so I want to expose an item that can be used externally to pass any type of error.
```rust
let connections_path = ConnectionsPath(conn_id.clone()).to_string().as_bytes().to_vec();

		if <Connections<T>>::contains_key(&connections_path) {
			let data = <Connections<T>>::get(&connections_path);
			let ret = ConnectionEnd::decode_vec(&data)
				.map_err(|_| Ics03Error::implementation_specific())?;

			Ok(ret)
		} else {
			Err(Ics03Error::connection_mismatch(conn_id.clone()))
		}
```

File: crates/ibc/src/core/ics04_channel/error.rs (hunk 1 above)
Comment (DaviRain-Su) [reply crates/ibc/src/core/ics04_channel/error.rs:355]: > We already have the [`AppModule` error](https://github.com/cosmos/ibc-rs/blob/0c5eb875f77f0de3360e58db957e8814aabbee60/crates/ibc/src/core/ics04_channel/error.rs#L343) for IBC applications.
> 
> What is the intended use of this error?

yet, for Application impl is ok, but for ics02, 03, 04 interface to bad

**Review by plafer — APPROVED** [review]
Thank you @DaviRain-Su!

**Pull request merged**